packages/eslint-plugin/src/rules/no-duplicate-type-constituents.ts

```diff
@@ -50,18 +51,13 @@ const isSameAstNode = (actualNode: unknown, expectedNode: unknown): boolean => {
     ) {
       return false;
     }
-    if (
-      actualNodeKeys.some(
-        actualNodeKey =>
-          !isSameAstNode(
-            actualNode[actualNodeKey as keyof typeof actualNode],
-            expectedNode[actualNodeKey as keyof typeof expectedNode],
-          ),
-      )
-    ) {
-      return false;
-    }
-    return true;
+    return !actualNodeKeys.some(
+      actualNodeKey =>
+        !isSameAstNode(
+          actualNode[actualNodeKey as keyof typeof actualNode],
+          expectedNode[actualNodeKey as keyof typeof expectedNode],
+        ),
+    );
```

Comment:
These changes don't seem to be related to the issue this PR solves?

Reply:
I thought it was minor, so I went ahead and fixed it.
However, as you mentioned, it's unrelated to this PR, so I reverted it.
[1560c83](https://github.com/typescript-eslint/typescript-eslint/pull/9600/commits/1560c83b826f50daace5593acda28bbb307cae77)